packages/transport/src/modules/sync/externalEventProcessors/ExternalEventProcessor.ts

```diff
@@ -0,0 +1,20 @@
+import { EventBus } from "@js-soft/ts-utils";
+import { AccountController } from "../../accounts/AccountController";
+import { Message } from "../../messages/local/Message";
+import { Relationship } from "../../relationships/local/Relationship";
+import { BackboneExternalEvent } from "../backbone/BackboneExternalEvent";
+import { ChangedItems } from "../ChangedItems";
+
+export type ExternalEventProcessorConstructor = new (eventBus: EventBus, changedItems: ChangedItems, accountController: AccountController) => ExternalEventProcessor;
+
+export abstract class ExternalEventProcessor {
+    public constructor(
+        protected readonly eventBus: EventBus,
+        protected readonly changedItems: ChangedItems,
+        protected readonly accountController: AccountController
+    ) {}
+    public abstract execute(externalEvent: BackboneExternalEvent): Promise<Message | Relationship | undefined>;
+    protected get address(): string {
```

Comment:
I think the naming `ownAddress` was way better also, please add a newline between the getter and execute method